src/Discord.Net.Rest/Extensions/EntityExtensions.cs

```diff
@@ -1,13 +1,26 @@
-﻿using System.Collections.Immutable;
+using System.Collections.Immutable;
 using System.Linq;
+using System.Threading.Tasks;
 
 namespace Discord.Rest
 {
     internal static class EntityExtensions
     {
-        public static GuildEmote ToEntity(this API.Emoji model)
+        public static GuildEmote ToEntity(this API.Emoji model, IGuild guild)
```

Comment:
Since `ToEntity()` is internal and all the public callsites are already async (I think), it doesn't seem like a breaking change to make this async as well.

Reply:
https://github.com/Chris-Johnston/Discord.Net/blob/322be8a31a6bb67f2d6451a25a73634ce7fe11fa/src/Discord.Net.WebSocket/Entities/Guilds/SocketGuild.cs#L340

https://github.com/Chris-Johnston/Discord.Net/blob/322be8a31a6bb67f2d6451a25a73634ce7fe11fa/src/Discord.Net.WebSocket/Entities/Guilds/SocketGuild.cs#L412

I've found these two usages that aren't within an async function. Given that I'd have to update all occurrences if I made `ToEntity` async anyways, maybe for these two cases I could use the `Result` property.

I'd like to get your opinion on this.